apps/designer/app/designer/features/style-panel/shared/property-name.tsx

```diff
@@ -14,8 +15,126 @@ import {
   Separator,
 } from "@webstudio-is/design-system";
 import { UndoIcon } from "@webstudio-is/icons";
+import type { Instance } from "@webstudio-is/react-sdk";
+import { utils } from "@webstudio-is/project";
 import { isFeatureEnabled } from "~/shared/feature-flags";
-import { getStyleSource, StyleInfo } from "./style-info";
+import { useSelectedInstanceData } from "~/designer/shared/nano-states";
+import { useBreakpoints, useRootInstance } from "~/shared/nano-states";
+import { type StyleInfo, type StyleSource, getStyleSource } from "./style-info";
+
+const PropertyPopoverContent = ({
+  properties,
+  style,
+  styleSource,
+  onReset,
+}: {
+  properties: StyleProperty[];
+  style: StyleInfo;
+  styleSource: StyleSource;
+  onReset: () => void;
+}) => {
+  const [breakpoints] = useBreakpoints();
+  const [rootInstance] = useRootInstance();
+  const [selectedInstanceData] = useSelectedInstanceData();
+  const selectedInstanceId = selectedInstanceData?.id;
+
+  // @todo replace with normalized data access
+
+  const breakpointsMap = useMemo(() => {
+    const breakpointsMap: Record<string, Breakpoint> = {};
+    for (const breakpoint of breakpoints) {
+      breakpointsMap[breakpoint.id] = breakpoint;
+    }
+    return breakpointsMap;
+  }, [breakpoints]);
+
+  const instancesMap = useMemo(() => {
+    const instancesMap: Record<string, Instance> = {};
```

Comment:
why not actual Map?

Reply:
actually looking how its being used
1. I think this map seems like an unnecessary optimization because depth won't be hundreds of items, so you could just lookup the array 
2. do we really need useMemo if getInstancePath is gonna be fast? we plan to optimize it with the new tree abstraction we planed, so building it should be very fast and this content is only computed when visible ...

Reply:
We won't need this code at all after normalization

Reply:
more reasons to not have it now :)